watch.go

```diff
@@ -103,48 +121,54 @@ func (fb *Firebase) Watch(notifications chan Event) error {
 			// 		event: put
 			// 		data: {"path":"/","data":{"foo":"bar"}}
 
-			var evt []byte
-			var dat []byte
-			isPrefix := true
-			var result []byte
-
-			// For possible results larger than 64 * 1024 bytes (MaxTokenSize)
-			// we need bufio#ReadLine()
-			// 1. step: scan for the 'event:' part. ReadLine() oes not return the \n
-			// so we have to add it to our result buffer.
-			evt, isPrefix, scanErr = scanner.ReadLine()
+			payload := make([]string, 2)
+			scanErr = func(payload []string) error {
```

Comment:
The logic inside of the closure is a little confusing. I like the idea of enforcing that we should only ever get two lines from the connection. What about something like

```go
var (
	lineCount int
	line      string
)
line, scanErr = scanner.ReadString('\n')
for ; lineCount != len(payload); line, scanErr = scanner.ReadString('\n') {
	if scanErr != nil {
		break scanning
	}

	line = strings.Trim(line, " \r\n")
	if len(line) == 0 { // empty line
		scanErr = errors.New("Bad formated body")
		break scanning
	}
	payload[lineCount] = line
	lineCount++
}
```